source/common/network/address_impl.cc

```diff
@@ -64,6 +64,70 @@ int Ipv4Instance::socket(SocketType type) const {
   return ::socket(AF_INET, flagsFromSocketType(type), 0);
 }
 
+std::string Ipv6Instance::Ipv6Helper::address() const {
+  return std::string(reinterpret_cast<const char*>(address_.sin6_addr.s6_addr),
+                     sizeof address_.sin6_addr.s6_addr);
+}
+
+uint32_t Ipv6Instance::Ipv6Helper::port() const {
+  return ntohs(address_.sin6_port);
+}
+
+std::string Ipv6Instance::Ipv6Helper::makeFriendlyAddress() const {
+  char str[INET6_ADDRSTRLEN];
+  if (nullptr == inet_ntop(AF_INET6, &address_.sin6_addr, str, INET6_ADDRSTRLEN)) {
+    // TODO: Consider pulling this out of ctor flow and into a static method used to construct
+    // instances, with the aim of being able to avoid exceptions in ctors. Discuss with Matt.
+    return "";
+  }
+  return str;
+}
+
+Ipv6Instance::Ipv6Instance(const sockaddr_in6& address) : InstanceBase(Type::Ip) {
+  ip_.ipv6_.address_ = address;
+  ip_.friendly_address_ = ip_.ipv6_.makeFriendlyAddress();
+  friendly_name_ = fmt::format("[{}]:{}", ip_.friendly_address_, ip_.port());
+}
+
+Ipv6Instance::Ipv6Instance(const std::string& address) : Ipv6Instance(address, 0) {}
+
+Ipv6Instance::Ipv6Instance(const std::string& address, uint32_t port) : InstanceBase(Type::Ip) {
+  memset(&ip_.ipv6_.address_, 0, sizeof(ip_.ipv6_.address_));
+  ip_.ipv6_.address_.sin6_family = AF_INET;
+  ip_.ipv6_.address_.sin6_port = htons(port);
+
+  // TODO: Consider pulling this out of ctor flow and into a static method used to construct
```

Comment:
@jamessynge did you forget to push a commit somewhere? You mention addressing comments in a commit below but I don't see a change here.